File: 2492/solution.java
class Solution {
    Set<Integer> vis = new HashSet<>();
    int res = Integer.MAX_VALUE;
    Map<Integer,List<int[]>> edges = new HashMap<>();
    
    public void dfs(int i){
        if(vis.contains(i))
            return;
        vis.add(i);
        for(int[] edge: edges.get(i)){
            res = Math.min(res, edge[1]);
            dfs(edge[0]);
        }
    }


    public int minScore(int n, int[][] roads) {
        for(int[] road: roads){
            edges.computeIfAbsent(road[0], k -> new ArrayList<>()).add(new int[]{road[1], road[2]});
            edges.computeIfAbsent(road[1], k -> new ArrayList<>()).add(new int[]{road[0], road[2]});

        }
        dfs(1);
        return res;
    }
}

File: 2492/solution.py
class Solution:
    def minScore(self, n: int, roads: List[List[int]]) -> int:
        adj = defaultdict(list)
        for src, nei, dist in roads:
            adj[src].append((nei,dist))
            adj[nei].append((src,dist))
        res = float('inf')
        vis = set()
        def dfs(i):
            if i in vis:
                return 
            vis.add(i)
            nonlocal res
            for nei, dist in adj[i]:
                res = min(res,dist)
                dfs(nei)
        dfs(1)
        return res
            
